### ISSUE TITLE
Is there a way to test other loss functions during training?

### ISSUE THREAD

**Issue by diogofd8** [mikel-brostrom/Yolov5_DeepSort_Pytorch#92]
Hello!
I felt curious about the training results using, for example, a cross entropy loss function (torch.nn.CrossEntropyLoss). However I couldn't manage to decipher what's being done with the BCE loss function in "loss.py".

Normally it would go like this:
'''
input would be a matrix with all the classes and their probabilities
target would be a matrix with zeros in all classes except the "target" one which would have a 1.

lossfunc = torch.nn.CrossEntropyLoss()
loss_value = loss(input, target)
'''

I've tried making some prints to figure out what goes in the BCE function and how the loss value is obtained but my python knowledge lacking. Do you mind giving me some tips?

**Comment by mikel-brostrom** [comment]
I recommend you to check the official PyTorch documentation of BCE. You can find its documentation [here](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) and its source code [here](https://pytorch.org/docs/stable/_modules/torch/nn/modules/loss.html#CrossEntropyLoss)
